src/DependencyManager.ts

```diff
@@ -0,0 +1,135 @@
+import { resolve } from 'path';
+import { readFile, run } from './util';
+import { coerce } from 'semver';
+
+export const LATEST_VERSION = 4;
+
+interface Dependency {
+	name: string;
+	version: string;
+	major: number;
+	minor: number;
+	patch: number;
+	isDevDependency: boolean;
+}
+
+interface PackageJson {
+	name: string;
+	version: string;
+	main?: string;
+	author?: string;
+	typings?: string;
+	license?: string;
+	homepage?: string;
+	repository?: { type: string; url: string };
+	scripts?: { [name: string]: string };
+	dependencies?: { [name: string]: string };
+	devDependencies?: { [name: string]: string };
+}
+
+export class DependencyManager {
+	private json: PackageJson;
+	private loadPromise: Promise<PackageJson>;
+	path: string;
+
+	constructor(path: string = resolve(process.cwd(), 'package.json')) {
+		this.setPackagePath(path);
+	}
+
+	private getPackageData(redownload?: boolean): Promise<PackageJson> {
```

Comment:
Why not just use `readFileSync`?

Reply:
With async/await isn't is just as easy and more flexible to use promises?

Reply:
Good point. I should've just used that, especially since I forgot Node 6 doesn't have a `promisify` method in util 😭.

Reply:
I removed it and went to readFileSync because there was a number of issues that this actually solves and simplifies. But yes, that was my original thinking.